pkg/ccn-lite/Makefile

```diff
@@ -3,40 +3,43 @@ PKG_URL=https://github.com/cn-uofbasel/ccn-lite/
 PKG_VERSION=e37b02c5cb20e9acccea2394be40f7e570a66a4b
 PKG_LICENSE=ISC
 
-.PHONY: all ..cmake_version_supported
+include $(RIOTBASE)/pkg/pkg.mk
 
-CMAKE_MINIMAL_VERSION = 3.6.0
+.PHONY: ..cmake_version_supported
 
+CMAKE_MINIMAL_VERSION = 3.6.0
 
 RIOT_CFLAGS = $(INCLUDES)
 
 ifeq (llvm,$(TOOLCHAIN))
   RIOT_CFLAGS += -Wno-char-subscripts
 endif
 
-TOOLCHAIN_FILE=$(PKG_BUILDDIR)/xcompile-toolchain.cmake
+TOOLCHAIN_FILE = $(PKG_BUILDDIR)/xcompile-toolchain.cmake
 
-all: $(PKG_BUILDDIR)/src/Makefile
-	$(MAKE) -C $(PKG_BUILDDIR)/src && \
-	cp $(PKG_BUILDDIR)/src/lib/libccnl-riot.a $(BINDIR)/ccn-lite.a
+all: $(BINDIR)/ccn-lite.a
 
-$(PKG_BUILDDIR)/src/Makefile: $(TOOLCHAIN_FILE)
-	cd $(PKG_BUILDDIR)/src && \
-	cmake -DCMAKE_TOOLCHAIN_FILE=$(TOOLCHAIN_FILE) \
-		  -DCCNL_RIOT=1 -DRIOT_CFLAGS="$(RIOT_CFLAGS)" -DBUILD_TESTING=OFF .
-$(TOOLCHAIN_FILE): git-download
-	$(RIOTTOOLS)/cmake/generate-xcompile-toolchain.sh > $(TOOLCHAIN_FILE)
+$(BINDIR)/ccn-lite.a: $(PKG_BUILDDIR)/bin/lib/libccnl-riot.a
+	cp $< $@
+
+$(PKG_BUILDDIR)/bin/lib/libccnl-riot.a: $(PKG_BUILDDIR)/bin/Makefile
+	$(MAKE) -C $(PKG_BUILDDIR)/bin clean
```

Comment:
The fact that we are cleaning here casuses `ccn-lite` tobe recompiled every time.

Reply:
If I remove the clean step, it still works and doesn't recompile. But I don't know the reason it was added in the first place (in #11553).

Reply:
For ccn-lite I get a different behaviour than expected,  when `make clean` is called the pkg is rebuilt which is not eh case in master.

Reply:
Hmm, not sure if I follow you. If I start clean and compile, the package is built. When I then recompile, it is much faster (yes, it invokes cmake but that also notices the file is the same and doesn't recompile).

```
Building application "ccn-lite-relay" for "native" with MCU "native".                                                                                                                                                                                                                                                                                                                           make[1]: Nothing to be done for 'prepare'.                                                                                                                                                      "make" -C /home/basilfx/RIOT/RIOT/pkg/ccn-lite                                                                                                                                                  /home/basilfx/RIOT/RIOT/dist/tools/cmake/generate-xcompile-toolchain.sh > /home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/xcompile-toolchain.cmake                      cmake -B/home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/bin -H/home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/src \                                   -DCMAKE_TOOLCHAIN_FILE=/home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/xcompile-toolchain.cmake \                                                                       -DCCNL_RIOT=1 -DRIOT_CFLAGS="-I/home/basilfx/RIOT/RIOT/core/include -I/home/basilfx/RIOT/RIOT/drivers/include -I/home/basilfx/RIOT/RIOT/sys/include -I/home/basilfx/RIOT/RIOT/boards/native/include -DNATIVE_INCLUDES -I/home/basilfx/RIOT/RIOT/boards/native/include/ -I/home/basilfx/RIOT/RIOT/core/include/ -I/home/basilfx/RIOT/RIOT/drivers/include/ -I/home/basilfx/RIOT/RIOT/cpu/native/include -I/home/basilfx/RIOT/RIOT/sys/include -I/home/basilfx/RIOT/RIOT/cpu/native/include -I/home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/src/ccnl-riot/include -I/home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/src/ccnl-core/include -I/home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/src/ccnl-pkt/include -I/home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/src/ccnl-fwd/include -I/home/basilfx/RIOT/RIOT/sys/posix/include" -DBUILD_TESTING=OFF                                  -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/bin
make -C /home/basilfx/RIOT/RIOT/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/bin
[ 35%] Built target ccnl-core
[ 41%] Built target ccnl-pkt
[ 48%] Built target ccnl-fwd
[100%] Built target ccnl-riot
"make" -C /home/basilfx/RIOT/RIOT/boards/native
...
```

Reply:
I get the output you have on `master` or on this PR when I do not clean, if I do `make clean all` on this PR It recompiles:

```
make -C /home/francisco/workspace/RIOT2/examples/ccn-lite-relay/bin/pkg/native/ccn-lite/bin
Scanning dependencies of target ccnl-core
[  1%] Building C object ccnl-core/CMakeFiles/ccnl-core.dir/src/ccnl-array.c.obj
[  2%] Building C object ccnl-core/CMakeFiles/ccnl-core.dir/src/ccnl-buf.c.obj
[  4%] Building C object ccnl-core/CMakeFiles/ccnl-core.dir/src/ccnl-callbacks.c.obj
[  5%] Building C object ccnl-core/CMakeFiles/ccnl-core.dir/src/ccnl-content.c.obj
[  7%] Building C object ccnl-core/CMakeFiles/ccnl-core.dir/src/ccnl-crypto.c.obj
[  8%] Building C object ccnl-core/CMakeFiles/ccnl-core.dir/src/ccnl-dump.c.obj
```

Reply:
Isn't that expected behaviour then?

Reply:
Yes, I was having trouble understanding the difference with master, but its because it was not cleaning the pkg properly in master (`clean -f` and not `clean -xdf`)

Reply:
> If I remove the clean step, it still works and doesn't recompile. But I don't know the reason it was added in the first place (in #11553).

I don't see why the clean is needed. I would just remove.

Reply:
The clean step has been removed.